src/components/menu/header-messages.jsx

```diff
@@ -77,6 +101,11 @@ export class HeaderMessagesUnconnected extends React.Component {
         className="notebook-header-messages-container"
       >
         {content}
+        <LoginModal
+          visible={this.state.showLoginModal}
+          onClose={this.hideLoginModal}
+          login={this.props.login}
```

Comment:
The login modal doesn't yet disappear when you click on login. I would create a helper function called `login` which hides the modal (or calls `this.hideLoginModal`) and then calls `this.props.login`.